AssemblerLib/ComputeSparsityPattern.cpp

```diff
@@ -0,0 +1,50 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2015, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "MeshLib/NodeAdjacencyTable.h"
+#include "LocalToGlobalIndexMap.h"
+
+#include "ComputeSparsityPattern.h"
+
+namespace AssemblerLib
+{
+
+SparsityPattern
+computeSparsityPattern(LocalToGlobalIndexMap const& dof_table,
+        MeshLib::Mesh const& mesh
+        )
+{
```

Comment:
It seems that this sparsity pattern is only for the case of indexing the entries of the stiffness matrix by component. If the global indexing is performed by location, the  sparsity pattern is made up of a number of same patches that are determined by NodeAdjacencyTable solely.